syntax = "proto3";

package NKikimrGraph;

option java_package = "ru.yandex.kikimr.proto";

message TMetric {
    string Name = 1;
    optional double Value = 2;
}

message TEvSendMetrics {
    repeated TMetric Metrics = 1;
    optional uint64 Time = 2; // for testing purposes only
}

message TEvGetMetrics {
    optional uint64 TimeFrom = 1;
    optional uint64 TimeTo = 2;
    repeated string Metrics = 3;
    optional uint32 MaxPoints = 4;
}

message TMetricData {
    repeated double Values = 1 [packed = true];
}

message TEvMetricsResult {
    repeated uint64 Time = 1 [packed = true];
    repeated TMetricData Data = 2;
    string Error = 3;
}
